src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -708,6 +713,7 @@ await _logger.RunWithProgressAsync(
                         _status = NuGetOperationStatus.Succeeded;
                     }
                     _auditResultCachingService.LastAuditCheckResult = _auditCheckResult;
+                    _didNewAuditCheckRun = true;
```

Comment:
Does this ever need to be set to `false`?

Reply:
No, defaults to false and I only set it to true when restore actually runs.